pkg/pillar/base/logobjecttypes.go

```diff
@@ -0,0 +1,291 @@
+// Copyright (c) 2020 Zededa, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package base
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"github.com/satori/go.uuid"
+	log "github.com/sirupsen/logrus"
+)
+
+// LogObjectType : Predefined log object types
+type LogObjectType string
+
+// The following is NOT an extensive list.
+// Intent is to create a reference list for engineers to get an idea of its usage.
+const (
+	UnknownLogType         LogObjectType = ""
+	LogType                LogObjectType = "log"
+	ImageLogType           LogObjectType = "image"
+	RelationLogType        LogObjectType = "relation"
```

Comment:
At the top level I think we have two types: objectLogType and relationLogType.
The relatioLogType then has a fromType/fromKey which can be e.g., AppInstanceLogType.

Thus I think we need to separate the LogType from the LogObjectType to make the relation fit in

Reply:
done.